src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -161,6 +161,11 @@ public function addNamePrefix(string $prefix)
         $prefixedRoutes = array();
 
         foreach ($this->routes as $name => $route) {
+            if (null !== $route->getDefault('_canonical_route')) {
+                $prefixedRouteName = $prefix.$route->getDefault('_canonical_route');
+                $route->setDefault('_canonical_route', $prefixedRouteName);
```

Comment:
I'd propose to move the added block after the line blow, doing this:
```php
            if (null !== $name = $route->getDefault('_canonical_route')) {
                $route->setDefault('_canonical_route', $prefix.$name);
            }
```

Reply:
Thanks for your review, I updated code following your advice